waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -100,7 +100,7 @@ type OnBeforeLoginHookParams = {
   /**
    * User that is trying to log in.
   */
-    user: Awaited<ReturnType<typeof findAuthWithUserBy>>['user']
+    user: NonNullable<Awaited<ReturnType<typeof findAuthWithUserBy>>>['user']
```

Comment:
We have a `FindAuthWithUserResult` type that we didn't export - we should export it and use it instead of the return type 👍 

`NonNullable` came into play because `findAuthWithUserBy` had the wrong return type of `User` and not `User | null` which is logical - user maybe doesn't exist in the DB.

Reply:
Updated to use `FindAuthWithUserResult` 👍